build.sh

```diff
@@ -14,6 +14,47 @@ function hasArg {
     (( NUMARGS != 0 )) && (echo " ${ARGS} " | grep -q " $1 ")
 }
 
+if hasArg --build-cuvs-from-source; then
+
+    CUR_DIR=$(pwd)
+    # Get cmake locally if not found
+    if ! cmake; then
+        echo "==> cmake not found. Downloading..."
+        CMAKE_URL="https://github.com/Kitware/CMake/releases/download/v4.1.1/cmake-4.1.1-linux-x86_64.tar.gz"
+        wget $CMAKE_URL
+        tar -xvf cmake-4.1.1-linux-x86_64.tar.gz
+        CMAKE_HOME="cmake-4.1.1-linux-x86_64"
+        export PATH="$CUR_DIR/$CMAKE_HOME/bin:$PATH"
+        echo "PATH: $PATH"
+    fi
+
+    unset LD_LIBRARY_PATH
+    CUVS_REPO="https://github.com/rapidsai/cuvs.git"
+    CUVS_DIR="cuvs"
+    BUILD_DIR=$CUVS_DIR/cpp/build
+
+    # checkout cuvs
+    if [ ! -d "$CUVS_DIR" ]; then
+        echo "==> Current working directory is: $CUR_DIR, checking out cuvs repository into $CUR_DIR..."
+        RV=$(git clone $CUVS_REPO $CUVS_DIR)
+        if [ "$RV" -ne 0 ]; then
+            echo "==> cuvs checkout failed."
+            exit "$RV"
+        fi
+    fi
+
+    # build cuvs
+    echo "==> Building cuvs"
+    if ! ./$CUVS_DIR/build.sh libcuvs; then
+        echo "==> cuvs build.sh returned non-zero."
+        exit "$RV"
+    fi
```

Comment:
This part doesn't look quite right.  Consider something like:

```bash
    RV=$(./$CUVS_DIR/build.sh libcuvs)
    if [ "$RV" -ne 0 ]; then
        echo "==> cuvs build.sh returned non-zero: $RV"
        exit "$RV"
    fi
```

Without this, `RV` would not have been defined by this point.